internal/errors/errors.go

```diff
@@ -0,0 +1,162 @@
+package errors
+
+import (
+	base "errors"
+	"fmt"
+	"reflect"
+	"runtime"
+)
+
+// The maximum number of stack frames on any error.
+const maxStackDepth = 50
+
+// interop with pkg/errors
+type causer interface{ Cause() error }
+
+// traced is an error with an attached stack trace
+type traced struct {
+	err    error
+	stack  []uintptr
+	frames []stackFrame
+	prefix string
+}
+
+func (e *traced) Error() string {
+	msg := e.err.Error()
+	if e.prefix != "" {
+		msg = fmt.Sprintf("%s: %s", e.prefix, msg)
+	}
+	return msg
+}
+
+func (e *traced) Unwrap() error {
+	return e.err
+}
+
+func (e *traced) String() string {
+	return e.TypeName() + " " + e.Error()
+}
+
+// Trace returns the call stack formatted the same way that go does in runtime/debug.Stack()
+func (e *traced) Trace() string {
+	trace := ""
+	for _, frame := range e.StackFrames() {
+		trace += frame.String()
+	}
+	return trace
+}
+
+// StackFrames returns an array of frames containing information about the stack.
+// This function must exist with this name for Sentry to report the correct trace.
+// See extractReflectedStacktraceMethod() in github.com/getsentry/sentry-go/stacktrace.go
+func (e *traced) StackFrames() []stackFrame {
+	if e.frames == nil {
+		e.frames = make([]stackFrame, len(e.stack))
+		for i, pc := range e.stack {
+			e.frames[i] = *newStackFrame(pc)
+		}
+	}
+
+	return e.frames
+}
+
+// FullTrace returns a string that contains both the error message and the call stack.
+func (e *traced) FullTrace() string {
+	return e.String() + "\n" + e.Trace()
+}
+
+// TypeName returns the type this error. e.g. *errors.stringError.
+func (e *traced) TypeName() string {
+	return reflect.TypeOf(e.err).String()
+}
+
+//// New replaces builtin errors.New()
+//func New(text string) error {
+//	return Err(text)
+//}
+
+// Err returns an error with stack trace
+func Err(e interface{}, fmtParams ...interface{}) error {
```

Comment:
Why not call it Wrap? For drop-in compatibility with `pkg/errors`, if not to avoid `errors.Err(err)`

Reply:
it doesn't just wrap errors, but creates new errors as well. You can give it a string or an error. And sometimes it doesn't wrap or create an error, just returns the existing error if it's already one of ours.

I'm open to another name. I couldn't think of another one.